src/controller.rs

```diff
@@ -177,7 +198,8 @@ impl Controller for FoxBox {
     }
 
     fn get_http_root_for_service(&self, service_id: String) -> String {
-        format!("http://{}:{}/services/{}/", self.hostname, self.http_port, service_id)
+        let scheme = if self.get_tls_enabled() { "https" } else { "http" };
+        format!("{}://{}:{}/services/{}/", scheme , self.hostname, self.http_port, service_id)
```

Comment:
Yes, you're either using TLS without any configured certs, in which case TLS won't work nor will http, or you're using plain HTTP.